objects/conversion.go

```diff
@@ -138,3 +139,123 @@ func ToByteSlice(o Object) (v []byte, ok bool) {
 	//ok = false
 	return
 }
+
+// ToMap will try to convert object o to map[string]interface value.
+func ToMap(o Object) (v map[string]interface{}, ok bool) {
+	v = make(map[string]interface{})
+	switch o := o.(type) {
+	case *Map:
+		for key, value := range o.Value {
+			v[key] = objectToInterface(value)
+		}
+		ok = true
+	}
+
+	//ok = false
+	return
+}
+
+// ToSlice will try to convert object o to []interface{} value.
+func ToSlice(o Object) (v []interface{}, ok bool) {
+	switch o := o.(type) {
+	case *Array:
+		for _, value := range o.Value {
+			v = append(v, objectToInterface(value))
+		}
+		ok = true
+	}
+
+	//ok = false
+	return
+}
+
+// objectToInterface attempts to convert an object o to an interface{} value
+func objectToInterface(o Object) (res interface{}) {
+	switch o := o.(type) {
+	case *Int:
+		res, _ = ToInt(o)
```

Comment:
Should be `res = o.Value` as `o` is already known to be `*Int`. Same for other primitive type cases here.